### PR TITLE
feat, added Notifier which implements basic fuctions

### LINKED ISSUE #8: Create Notifier
Notifier receives commit hash and its status from Builder. Then it reads [GitHub auth token](https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps) from a file and [sends](https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#create-a-commit-status) a commit status update to GitHub.

### PR DESCRIPTION
**Pull request by dengty1998**

### CODE REVIEW

File: src/main/java/assignment/Notifier.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public static JSONObject setJson(JSONObject commitStatus) {
+        String testResult = "";
+        String state;
+        String targetUrl = "";
+        String description;
+        String context = "";
+
+        if (testResult.equals("buildFailed")) {
+            state = "success";
+            description = "Succeeded, passed the tests";
```
Comment (aversey) [inline src/main/java/assignment/Notifier.java]: Looks like "Failed, compilation failed" needs to be here. Success is to be done in a separate branch.

**Review by aversey — CHANGES_REQUESTED** [review]
"set" in the names of functions doesn't seem to be the right word. Don't forget to load secret token from a file (on path "./token"?) and to provide some way for passing the build status and related commit ID to your code (maybe through constructor or method "notify").

Overall, good work, keep up!

File: src/main/java/assignment/Notifier.java (hunk 1 above)
Comment (dengty1998) [reply src/main/java/assignment/Notifier.java]: > Looks like "Failed, compilation failed" needs to be here. Success is to be done in a separate branch.

Ah... I just mistype some code lines. Will fix this

**Comment by dengty1998** [comment]
> "set" in the names of functions doesn't seem to be the right word. Don't forget to load secret token from a file (on path "./token"?) and to provide some way for passing the build status and related commit ID to your code (maybe through constructor or method "notify").
> 
> Overall, good work, keep up!

In fact, I just don't quite sure about how to get a token(and username) and don't know what kind of data will be sent to me. That's why I create some parameter but leave them empty......:
        String testResult = "";
        String state;
        String targetUrl = "";
        String description;
        String context = "";
        String urlString = "";
        String shaString = "";
        String userString = "";
        String tokenString = "";
And I am not sure about code block in try{} either. Because I have never written a similar function.
So I feel there are still many things that need to be modified...

**Review by aversey — APPROVED** [review]

**Review by JonasHUL — APPROVED** [review]
Well done!!

**Pull request merged**